crates/wash-cli/src/cmd/wit/build.rs

```diff
@@ -0,0 +1,83 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use clap::Args;
+use wash_lib::build::load_lock_file;
+use wash_lib::cli::{CommandOutput, CommonPackageArgs};
+use wash_lib::parser::load_config;
+
+use wasm_pkg_core::wit::{self};
+
+/// Arguments used for `wash wit build`
+#[derive(Debug, Args, Clone)]
+pub struct BuildArgs {
+    /// The directory containing the WIT files to build.
+    #[clap(short = 'd', long = "wit-dir", default_value = "wit")]
+    pub dir: PathBuf,
+
+    /// The name of the file that should be written. This can also be a full path. Defaults to the
+    /// current directory with the name of the package
+    #[clap(short = 'f', long = "file")]
+    pub output_file: Option<PathBuf>,
+
+    #[clap(flatten)]
+    pub common: CommonPackageArgs,
+
+    /// Path to the wasmcloud.toml file or parent folder to use for building
+    #[clap(short = 'p', long = "config-path")]
+    config_path: Option<PathBuf>,
+}
+
+/// Invoke `wash wit build`
+pub async fn invoke(
+    BuildArgs {
+        dir,
+        output_file,
+        common,
+        config_path,
+    }: BuildArgs,
+) -> anyhow::Result<CommandOutput> {
+    let client = common.get_client().await?;
+    // Attempt to load wasmcloud.toml. If it doesn't work, attempt to load wkg.toml
+    let wkg_config = if let Ok(proj) = load_config(config_path, Some(true)).await {
+        proj.package_config
+    } else {
+        wasm_pkg_core::config::Config::load().await?
+    };
+
+    let mut lock_file =
+        load_lock_file(std::env::current_dir().context("failed to get current directory")?).await?;
+
+    // Build the WIT package
+    let (pkg_ref, version, bytes) =
+        wit::build_package(&wkg_config, dir, &mut lock_file, client).await?;
+
+    // Resolve the output path
+    let output_path = match output_file {
+        Some(path) => path,
+        None => {
+            let mut file_name = pkg_ref.to_string();
+            if let Some(ref version) = version {
+                file_name.push_str(&format!("@{version}"));
```

Comment:
I think the `@` symbol is forbidden on windows. I got away wit separating things with `-` instead. This is something I need to fix in `wkg` as well so just thought I'd give you a heads up

Reply:
Note that this code was moved -- [I didn't actually change this!](https://github.com/wasmCloud/wasmCloud/blob/0af83f4148833b0c5187473b8572e093928aef22/crates/wash-cli/src/wit.rs#L112)

Also one thing I did notice is that there was no way to tell where the namespace started and the package ended -- can you also change that upstream while you're there? i.e. `test-components__testing__0.1.0` would be nicer to have than `test-components-testing-0.1.0`

Reply:
Here's a [new issue to track this ](https://github.com/wasmCloud/wasmCloud/issues/3799) when upstream changes

Reply:
Oh I figured this was me having mostly copy pasted from wkg. Was just giving a heads up